### ISSUE TITLE
false offense for Style/CommentedKeyword

### ISSUE THREAD

**Issue by chuck-john** [bbatsov/rubocop#4886]
## Expected behavior

I expect rubocop to allow literal octothorpes and interpolated strings in method arguments.

## Actual behavior

An offense was generated for an interpolated string in a method argument. The offense claims this is a comment, which triggered a violation of the `Style/CommentedKeyword` cop.

## Steps to reproduce the problem

1. Define a method with an octothorpe in a string as a default method argument.
1. Run `$ rubocop`
1. An offense similar to the screenshot below should get generated.

![rubocop v0 51 0 - style commentedkeyword](https://user-images.githubusercontent.com/1545623/31775117-52348cce-b4ad-11e7-9a3b-28f5b06b71b7.png)

## RuboCop version

0.51.0

**Comment by codebycliff** [comment]
I have another example of a false positive:

![image](https://user-images.githubusercontent.com/651946/31777168-027ae83a-b4b3-11e7-8450-ebf090d4be98.png)

**Comment by Drenmi** [comment]
Hm. I'm guessing this cop uses a regular expression instead of checking for comments outside the AST. 😅

**Comment by backus** [comment]
> Hm. I'm guessing this cop uses a regular expression instead of checking for comments outside the AST. 😅

We should add an internal affairs cop that bans this. Seems to be the source of a lot of bugs

**Comment by bbatsov** [comment]
@backus True. And I guess we don't review these things very carefully. 😆

**Comment by wurde** [comment]
Same problem here.

```ruby
app/helpers/bootstrap/alert_helper.rb:49:40: C: Do not place comments on the same line as the def keyword.
    def alert_link(name = nil, path = '#', options = {}, &block)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by reitermarkus** [comment]
Also here:

```ruby
def self.tap_paths(name, taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"])
```

**Comment by michniewicz** [comment]
same for `#` inside Regexp:
```
foo.rb:2:20: C: Do not place comments on the same line as the def keyword.
  def foo(type = /^#{self.bar}_/)
```
